### DISCUSSION TITLE
Page routing reloads whole pages

### DISCUSSION THREAD

**Discussion by waseem-netsolutions** [vercel/next.js#49592] [Help] [answered]
### Summary

When I navigate through the links, whole pages get reloaded and it seems _app get re-initialized. I have put some Modals in the custom app and they need to open initially. But they get open on every route change.

[(https://hcc2-fe-dev.hancocks.co.uk/) ](https://hcc2-fe-dev.hancocks.co.uk/)  -- For example if you go to create account , those modals open again

### Additional information
 We are using AWS ec2 and we have run app on port 8080 and our domian points to that port.
_No response_

**Comment by icyJoseph** [comment]
Could you give us some pseudo-code of what's going on in side `_app` specifically, what does the custom App component return?

**Reply by waseem-netsolutions** [reply]
```javascript
export default function App({ Component, ...rest }) {

  const getLayout = Component.getLayout || getDefaultLayout;
  const { store, props } = wrapper.useWrappedStore(rest);
  const { pageProps } = props;

  const queryClient = useMemo(() => new QueryClient({
    defaultOptions: {
      queries: {
        refetchOnWindowFocus: false,
        retry: false,
        refetchOnMount: false
      }
    }
  }), []);

  return (
    <>
      <QueryClientProvider client={queryClient}>
        <Hydrate state={pageProps.dehydratedQueryClient}>
          <ReduxProvider store={store}>
           // Modals are present in <AppInitialization/> component
            <AppInitialization />
            {getLayout(<Component {...pageProps} />)}
            <ReactQueryDevtools initialIsOpen={false} />
          </ReduxProvider>
        </Hydrate>
      </QueryClientProvider>
    </>
  )
}
```

**Reply by icyJoseph** [reply]
Not really what I was hoping for, I was expecting that you were recreating the component instance, but at first glance that's not what's going on here. 

I guess `guessLayout` is according to the docs right?

**Reply by icyJoseph** [reply]
One thing I saw, is that clicking your links, unloads the page. Why would that happen, that's super weird. It's like the JS or JSON for the pages is not loading, or something like that.

**Reply by icyJoseph** [reply]
I'm starting to think this has to be because _next/data calls are not working. They're all rejected. Do you have some kind of server rules setup?

**Reply by waseem-netsolutions** [reply, marked as answer]
**I removed the next-translate and next-translate-plugin. And everything just worked fine.** 

I was using it like this in next.config.js

This was not supposed to create any problems but apparently it did.
```javascript
const nextTranslate = require('next-translate-plugin')

module.exports = nextTranslate({
  reactStrictMode: true,
  sassOptions: {
    fiber: false
  },
})
```

**Reply by icyJoseph** [reply]
Aha, do you know which one exactly triggered the issue (or is it the same plugin)? Were you ever able to see if fail with a production build, locally? (`next build` and `next start` on your machine?)

**Reply by waseem-netsolutions** [reply]
It was working smoothly on my local machine. Once I deployed the build, it started behaving weirdly.

**Reply by icyJoseph** [reply]
oh well, I think you can select your comment as answer :) 🎉
